pkg/controller/jobs/appwrapper/appwrapper_controller.go

```diff
@@ -128,11 +130,28 @@ func (aw *AppWrapper) PodSets() ([]kueue.PodSet, error) {
 	}
 	podSets := make([]kueue.PodSet, len(podSpecTemplates))
 	for psIndex := range podSpecTemplates {
+		var podIndexLabel *string
+		var subGroupIndexLabel *string
+		var subGroupCount *int32
+		if annotation, ok := awPodSets[psIndex].Annotations[awutils.PodSetAnnotationTASPodIndexLabel]; ok {
+			podIndexLabel = &annotation
+		}
+		if annotation, ok := awPodSets[psIndex].Annotations[awutils.PodSetAnnotationTASSubGroupIndexLabel]; ok {
+			subGroupIndexLabel = &annotation
+		}
+		if annotation, ok := awPodSets[psIndex].Annotations[awutils.PodSetAnnotationTASSubGroupCount]; ok {
+			if count, err := strconv.Atoi(annotation); err == nil {
+				subGroupCount = ptr.To[int32](int32(count))
+			} else {
+				ctrl.Log.Error(err, fmt.Sprintf("Malformed %v annotation ignored", awutils.PodSetAnnotationTASSubGroupCount),
```

Comment:
```suggestion
				ctrl.Log.Error(err, "Malformed annotation ignored", "annotationKey", awutils.PodSetAnnotationTASSubGroupCount, "annotationValue", count),
```

The `ctrl.Log.Error()` has `keyAndValues`. So, it would be better to rely on it.